### PR TITLE
Add EEP A5-20-04

### PR DESCRIPTION
**Pull request by wallenium**

### CODE REVIEW

**Comment by mak-gitdev** [comment]
Hi @wallenium,

Thanks for your PR.

`FTS` and `TMPFC` fields values depend on the value of the `TS` field.
So I don't think we can support scaling at this level, otherwise the value will only be OK in one case instead of both.
I think we should handle scaling at HA level.

The direction 2 is also missing.

Could you rework your PR regarding the above comments ? I will be happy to integrate it once OK.

**Comment by wallenium** [comment]
Hi mak,
I added direction 2

For dir 1, was there anything wrong or was all ok? I would implement the rest in HA ;)

File: enocean/protocol/EEP.xml [hunk 1]
```diff
@@ -1372,6 +1372,177 @@
...
+              <max>100</max>
+            </scale>
+          </value>
+          <value shortcut="FTS" description="Feed Temperature OR Temperature Set Point" offset="8" size="8" unit="°C">
+            <range>
+              <min>0</min>
+              <max>255</max>
+            </range>
+            <scale>
+              <min>10</min>
```
Comment (mak-gitdev) [inline enocean/protocol/EEP.xml]: No scale should be defined here, as this is shared between two values using different scale values.

File: enocean/protocol/EEP.xml [hunk 2]
```diff
@@ -1372,6 +1372,177 @@
...
+              <max>30</max>
+            </scale>
+          </value>
+          <value shortcut="TMPFC" description="Room Temperature OR Failure Code" offset="16" size="8" unit="°C">
+            <range>
+              <min>0</min>
+              <max>255</max>
+            </range>
+            <scale>
```
Comment (mak-gitdev) [inline enocean/protocol/EEP.xml:1402]: Same comment as for `FTS`

File: enocean/protocol/EEP.xml [hunk 3]
```diff
@@ -1372,6 +1372,177 @@
...
+              <min>10</min>
+              <max>30</max>
+            </scale>
+          </value>
+          <value shortcut="TMPFC" description="Room Temperature OR Failure Code" offset="16" size="8" unit="°C">
```
Comment (mak-gitdev) [inline enocean/protocol/EEP.xml]: No unit because it is not OK when this field is used to send failure code.

File: enocean/protocol/EEP.xml [hunk 4]
```diff
@@ -1372,6 +1372,177 @@
...
+          </value>
+          <enum shortcut="MST" description="Measure Status" offset="24" size="1">
+            <item description="active" value="0" />
+            <item description="inactive" value="1" />
+          </enum>
+          <enum shortcut="STR" description="Status Request" offset="25" size="1">
+            <item description="No change" value="0" />
+            <item description="status request" value="1" />
+          </enum>
+          <enum shortcut="LRNB" description="LRN Bit" offset="28" size="1">
```
Comment (mak-gitdev) [inline enocean/protocol/EEP.xml]: No need to declare the learn bit. This is automatically handled by the library.

File: enocean/protocol/EEP.xml [hunk 5]
```diff
@@ -1372,6 +1372,177 @@
...
+            <item description="39 hrs" value="62" />
+            <item description="42 hrs (max)" value="63" />
+          </enum>
+          <enum shortcut="DSO" description="Display Orientation" offset="26" size="2">
+            <item description="0°" value="0" />
+            <item description="90°" value="1" />
+            <item description="180°" value="2" />
+            <item description="270°" value="3" />
+          </enum>
+          <enum shortcut="LRNB" description="LRN Bit" offset="28" size="1">
```
Comment (mak-gitdev) [inline enocean/protocol/EEP.xml]: Same comment for learn bit.

**Review by mak-gitdev — CHANGES_REQUESTED** [review]
Hi @wallenium,

Thanks again for your PR.
Please have a look at my comments.

File: enocean/protocol/EEP.xml (hunk 3 above)
Comment (wallenium) [reply enocean/protocol/EEP.xml]: done

File: enocean/protocol/EEP.xml (hunk 2 above)
Comment (wallenium) [reply enocean/protocol/EEP.xml:1402]: done

File: enocean/protocol/EEP.xml (hunk 1 above)
Comment (wallenium) [reply enocean/protocol/EEP.xml]: done

File: enocean/protocol/EEP.xml (hunk 4 above)
Comment (wallenium) [reply enocean/protocol/EEP.xml]: removed

**Comment by wallenium** [comment]
I added all changes.

File: enocean/protocol/EEP.xml [hunk 6]
```diff
@@ -1389,20 +1389,12 @@
               <min>0</min>
               <max>255</max>
             </range>
-            <scale>
```
Comment (mak-gitdev) [inline enocean/protocol/EEP.xml]: Sorry,
Just read the code of the EnOcean library and it seems that once `range` is defined `scale` must be defined as well and vice-versa.
So we should just add:
```xml
            <scale>
              <min>0</min>
              <max>255</max>
            </scale>
```

File: enocean/protocol/EEP.xml [hunk 7]
```diff
@@ -1389,20 +1389,12 @@
...
-              <max>30</max>
-            </scale>
           </value>
-          <value shortcut="TMPFC" description="Room Temperature OR Failure Code" offset="16" size="8" unit="°C">
+          <value shortcut="TMPFC" description="Room Temperature OR Failure Code" offset="16" size="8">
             <range>
               <min>0</min>
               <max>255</max>
             </range>
-            <scale>
```
Comment (mak-gitdev) [inline enocean/protocol/EEP.xml]: Same comment. Sorry.

**Review by mak-gitdev — CHANGES_REQUESTED** [review]
Last comments. Sorry

**Review by mak-gitdev — APPROVED** [review]
All good

**Pull request merged**